### ISSUE TITLE
Broken video in Firefox after Mesa update

### ISSUE THREAD

**Issue by jtojnar** [NixOS/nixpkgs#200592]
### Describe the bug

Videos played in Firefox appear to have broken chrominance:

![Example of broken video](https://user-images.githubusercontent.com/705123/201209912-cc286b97-953f-474f-8e0b-2158b32e5ee2.png)

### Steps To Reproduce
1. Open YouTube
2. Play random video

### Additional context

I have bisected it to interval 3c76783506e7301748f955abf787f3d7f317ee52 (good) and f994293d1eb8812f032e8919e10a594567cf6ef7 (bad), reverting Mesa to an older version with the following fixes it:

```nix
  system.replaceRuntimeDependencies =
    let
      nixpkgsfixed = builtins.fetchTarball "https://github.com/NixOS/nixpkgs/archive/3c76783506e7301748f955abf787f3d7f317ee52.tar.gz";
      mesa = pkgs.callPackage (nixpkgsfixed + "/pkgs/development/libraries/mesa") {
        inherit (pkgs.darwin.apple_sdk.frameworks) OpenGL;
        inherit (pkgs.darwin.apple_sdk.libs) Xplugin;
      };
    in
    builtins.map
      (output: {
        original = pkgs.mesa.${output};
        replacement = mesa.${output};
      })
      pkgs.mesa.outputs;
}}}
```

445887f636e75d6d057d8d46980d442d7f1cd02d appears to be a first bad commit out of https://github.com/NixOS/nixpkgs/commits/510109763ccb20645bbdaa577b0306ae18d795e7/pkgs/development/libraries/mesa/default.nix

It works properly when FIrefox is run with `--safe-mode`.

### Notify maintainers

cc @PedroHLC @LunNova  @vcunat 

### Metadata

Using nixos-unstable (093268502280540a7f5bf1e2a6330a598ba3b7d0)

**Comment by vcunat** [comment]
I'm using a little later mesa now (22.2.2) without experiencing the issue, but it may be depending on GPU or something else, too.

**Comment by vcunat** [comment]
`.3` is in staging, but the shortlog doesn't seem to contain anything specific to nouveau.  I'm not sure what to do about this.

**Comment by vcunat** [comment]
Actually, it sounds worth retrying, as this one could be related: https://gitlab.freedesktop.org/mesa/mesa/-/issues/5632

**Comment by jtojnar** [comment]
Looks like it is indeed fixed on 1462b5397b2c3b8d9891b41de131d0776fd4a63f. Thanks.
